src/main/scala/latis/service/hapi/Latis3Interpreter.scala

```diff
@@ -61,18 +70,23 @@ class Latis3Interpreter extends HapiInterpreter[IO] {
     // Header and format are handled by DataService.
     case DataRequest(id, minT, maxT, params, _, _) =>
       val ops: List[UnaryOperation] = List(
-        Selection(s"time >= $minT"),
-        Selection(s"time < $maxT")
+        Selection(id"time", GtEq, s"$minT"),
+        Selection(id"time", Lt, s"$maxT")
       ) ++ makeProjection(params)
 
       getDataset(id, ops)
   }
 
   override def writeData(data: T): Stream[IO, String] =
-    new CsvEncoder().encode(data)
+    CsvEncoder().encode(data)
 
   private def getDataset(id: String, ops: List[UnaryOperation]): IO[T] = IO {
-    val dataset: T = DatasetResolver.getDataset(id)
+    val ident = Identifier.fromString(id).getOrElse {
+      throw LatisException(s"Invalid Identifier: $id")
+    }
+    val dataset: T = catalog.findDataset(ident).unsafeRunSync().getOrElse {
+      throw LatisException(s"Unable to find dataset: $id")
+    }
     ops.foldLeft(dataset)(_.withOperation(_)).withOperation(new ToHapiTime)
   }
```

Comment:
This should be rewritten to not use `unsafeRunSync`.

Reply:
Done.